pkg/receive/multitsdb.go

```diff
@@ -563,7 +567,16 @@ func (t *MultiTSDB) startTSDB(logger log.Logger, tenantID string, tenant *tenant
 	reg := prometheus.WrapRegistererWith(prometheus.Labels{"tenant": tenantID}, t.reg)
 	reg = NewUnRegisterer(reg)
 
-	lset := labelpb.ExtendSortedLabels(t.labels, labels.FromStrings(t.tenantLabelName, tenantID))
+	initialLset := labelpb.ExtendSortedLabels(t.labels, labels.FromStrings(t.tenantLabelName, tenantID))
+
+	lset, err := t.extractTenantsLabels(tenantID, initialLset)
+	if err != nil {
+		t.mtx.Lock()
+		delete(t.tenants, tenantID)
```

Comment:
I am not sure we should be doing this. We probably just want to return the error and let the user fix the config.

Reply:
done